draft-ietf-quic-transport.md

```diff
@@ -2979,10 +2979,49 @@ packet.
 expectations about what implementations do with packets that have errors after
 valid frames? -->
 
+
+## Generating Acknowledgements {#generating-acks}
+
+Endpoints acknowledge all packets they receive and process. However, only
+ack-eliciting packets trigger the sending of an ACK frame.  Packets that are not
+ack-eliciting are only acknowledged when an ACK frame is sent for other reasons.
+
+When sending a packet for any reason, an endpoint should attempt to bundle an
+ACK frame if one has not been sent recently. Doing so helps with timely loss
+detection at the peer.  While loss detection and congestion control do not
+depend on receiving an immediate acknowledgment for packets that are sent,
+performance can be improved by acknowledging immediately in several cases, as
+discussed below.
+
 ### Sending ACK Frames
 
-An endpoint sends ACK frames to acknowledge packets it has received and
-processed.
+An ACK frame SHOULD be sent immediately upon receipt of a second ack-eliciting
+packet. Loss detection does not assume that the peer sends an ACK immediately on
+receiving a second ack-eliciting packet.
```

Comment:
These are both good and useful statements, but they sort of stand awkwardly next to each other as if they are strangers.

> An endpoint SHOULD immediately send an ACK frame when two ack-eliciting packets have been received without being acknowledged.  Though the algorithms in {{QUIC-RECOVERY}} do not depend on receivers following this guidance, frequent feedback improves loss and congestion response.

I tried to say something about the fact that this also provides some assurances about acknowledgement behaviour that senders could use when innovating with new algorithms, but it ended up being quite hard to do that without distracting too much from the core purpose of the section.  Also, the SHOULD got in the way: there is no real assurance, there we have to rely on the fact that this is a good recommendation that most deployments will likely respect.

Reply:
I want to do another larger pass over this text, but I'd like to keep it as is for now.

Reply:
I'm trying out something -- give me a few minutes

Reply:
Ok, a few changes, look at the last commit. I've moved some text around and I think I've captured the spirit of this more clearly.